Bug: `test_lldp.py` testlets fail with `Missing in lldpcli: ['Ethernet-Rec0']`

### Is it platform specific

generic

### Importance or Severity

Medium

### Description of the bug

Two new testlets were added in `test_lldp.py` in https://github.com/sonic-net/sonic-mgmt/pull/22420
`test_lldp_interfaces` and `test_lldp_interfaces_config_reload`.
But these two new tests don't consider if a platform has recycle ports to exclude them from the comparison with the LLDP output.
Hence the error:
```
>       pytest_assert(intf_status_filtered_for_lldpcli == lldpcli_interfaces_no_eth0,
                      "Interface mismatch between 'show interface status' (no PortChannels) and lldpcli{}. "
                      "Missing in lldpcli: {}, Extra in lldpcli: {}".format(
                          context, sorted(missing_in_lldpcli), sorted(extra_in_lldpcli)))
E       Failed: Interface mismatch between 'show interface status' (no PortChannels) and lldpcli. Missing in lldpcli: ['Ethernet-Rec0'], Extra in lldpcli: []
```

### Steps to Reproduce

Run `test_lldp_interfaces` or `test_lldp_interfaces_config_reload` on a SKU with recycle ports

### Actual Behavior and Expected Behavior

`test_lldp_interfaces` and `test_lldp_interfaces_config_reload` should exclude the recycle ports from the comparison with the lldp output

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/lldp/test_lldp.py
+++ b/tests/lldp/test_lldp.py
@@ -353,7 +353,9 @@ def verify_lldp_table(duthost, intf_status_output, test_name=""):
     # Filter intf_status_output: exclude PortChannel interfaces and admin down interfaces
     intf_status_filtered_for_lldp = {
         intf['interface'] for intf in intf_status_output
-        if not intf['interface'].startswith('PortChannel') and intf['admin'].lower() == 'up'
+        if not intf['interface'].startswith('PortChannel') and
+        not intf['alias'].startswith('Recirc') and
+        intf['admin'].lower() == 'up'
     }
 
     missing_in_lldp_table = intf_status_filtered_for_lldp - lldp_table_interfaces_no_eth0
@@ -436,12 +438,15 @@ def verify_lldpcli_interfaces(duthost, asic, intf_status_output, test_name=""):
             asic.asic_index, len(asic_ports)))
         intf_status_filtered_for_lldpcli = {
             intf['interface'] for intf in intf_status_output
-            if not intf['interface'].startswith('PortChannel') and intf['interface'] in asic_ports
+            if not intf['interface'].startswith('PortChannel') and
+            not intf['alias'].startswith('Recirc') and
+            intf['interface'] in asic_ports
         }
     else:
         intf_status_filtered_for_lldpcli = {
             intf['interface'] for intf in intf_status_output
-            if not intf['interface'].startswith('PortChannel')
+            if not intf['interface'].startswith('PortChannel') and
+            not intf['alias'].startswith('Recirc')
         }
 
     missing_in_lldpcli = intf_status_filtered_for_lldpcli - lldpcli_interfaces_no_eth0
@@ -507,7 +512,9 @@ def verify_lldpctl_facts(duthost, enum_frontend_asic_index, intf_status_output, 
     # Compare intf_status_output with lldpctl_facts interfaces (exclude PortChannels and admin down from intf_status)
     intf_status_filtered_for_lldpctl = {
         intf['interface'] for intf in intf_status_output
-        if not intf['interface'].startswith('PortChannel') and intf['admin'].lower() == 'up'
+        if not intf['interface'].startswith('PortChannel') and
+        not intf['alias'].startswith('Recirc') and
+        intf['admin'].lower() == 'up'
     }
 
     missing_in_lldpctl_facts = intf_status_filtered_for_lldpctl - lldpctl_facts_interfaces

